frontend/controller/actions/group.js

```diff
@@ -101,13 +122,63 @@ export default (sbp('sbp/selectors/register', {
         // handle Flowtype annotations, even though our .babelrc should make it work.
         distributionDate = dateToPeriodStamp(addTimeToDate(new Date(), 3 * DAYS_MILLIS))
       }
-      const message = await sbp('chelonia/out/registerContract', {
+      const message = await sbp('chelonia/withEnv', '', {
+        additionalKeys: {
+          [CSKid]: CSK,
+          [CEKid]: CEK,
+          [inviteKeyId]: inviteKey
+        }
```

Comment:
To summarize the call:

The plan to address the other comment about removing `withEnv` usage and DRY violations of repeatedly passing in `additionalKeys`, and to simplify the API, is to pass in `additionalKeys` once via `chelonia/configure`, anytime we want Chelonia to temporarily remember these keys.

So that includes calling `chelonia/configure` once in this function, and also, upon login when the user logs in, to temporarily remember the IPK stuff.